clusterloader2/testing/watch-list/config.yaml

```diff
@@ -0,0 +1,84 @@
+name: huge-secrets
+tuningSets:
+- name: Uniform10qps
+  qpsLoad:
+    qps: 10
+steps:
+- name: Create well known namespaces
+  phases:
+  # don't know why, but it has to be set
+  # otherwise I get some compilation errors
+  - namespaceRange: null
+    # this seems to be controlling the
+    # number of ns I want to create
+    replicasPerNamespace: 1
+    tuningSet: Uniform10qps
+    objectBundle:
+    - basename: huge-secrets
+      objectTemplatePath: "ns.yaml"
+- name: Create secrets
+  phases:
+  - namespaceRange:
+      min: 0
+      max: 0
+      basename: huge-secrets
+    replicasPerNamespace: 400
+    tuningSet: Uniform10qps
+    objectBundle:
+    - basename: huge-secret
+      objectTemplatePath: "secret.yaml"
+- name: Start measurements
+  measurements:
+    - Identifier: MemoryProfile
+      Method: MemoryProfile
+      Params:
+        action: start
+    - Identifier: WaitForControlledPodsRunning
+      Method: WaitForControlledPodsRunning
+      Params:
+        action: start
+        apiVersion: apps/v1
+        kind: Deployment
+        labelSelector: group = watch-list
+        operationTimeout: 60s
+- name: Start the secret informer
+  phases:
+    - namespaceRange:
+        min: 0
+        max: 0
+        basename: huge-secrets
+      replicasPerNamespace: 1
+      tuningSet: Uniform10qps
+      objectBundle:
+        - basename: huge-secret
+          objectTemplatePath: role.yaml
+        - basename: huge-secret
+          objectTemplatePath: roleBinding.yaml
+        - basename: huge-secret
+          objectTemplatePath: "deployment.yaml"
+          templateFillMap:
+            Replicas: 1
+- name: Wait for the secret informer pod to be running
+  measurements:
+    - Identifier: WaitForControlledPodsRunning
+      Method: WaitForControlledPodsRunning
+      Params:
+        action: gather
+- name: Measure the API server memory usage
+  measurements:
+    - Identifier: MemoryProfile
+      Method: MemoryProfile
+      Params:
+        action: gather
+- name: Delete well known namespaces
```

Comment:
How do I know that my test has finished?

Reply:
I'm not sure if now CL2 supports waiting for pod termination.

I think it may be easier to achieve desired effect in the current clusterloader model with:
* pod image are probing until they are kiled
* CL2 starts deployment, waits for it to be started
* Sleeps 5m
* CL2 deletes deployment, waits for all pods to be deleted

Reply:
Alternatively you can migrate to Jobs and use "WaitForFinishedJobs" https://github.com/kubernetes/perf-tests/blob/4da7097e35b13225ed05bf3dfcb842864c11fae4/clusterloader2/pkg/measurement/common/wait_for_jobs.go#L17-L17

Reply:
decided to take the first approach, added a sleep, seems to be working.